plugins/blip/basics/logic/datamunger.js

```diff
@@ -28,6 +28,42 @@ module.exports = function(bgClasses) {
   var classifiers = classifiersMkr(bgClasses);
 
   return {
+    calculateCarbStats: function(basicsData) {
+
+      var wizardEvents = basicsData.data.wizard.data;
+
+      var carbs = _.filter(wizardEvents, function(wizardEvent) {
+        return wizardEvent.carbInput && wizardEvent.carbInput > 0 ;
+      });
+
+      if (!_.isEmpty(carbs)) {
+
+        var start = carbs[0].normalTime;
+        if (start < basicsData.dateRange[0]) {
+          start = basicsData.dateRange[0];
+        }
+        var end = carbs[carbs.length - 1].normalTime;
+        if (end > basicsData.dateRange[1]) {
+          end = basicsData.dateRange[1];
+        }
+
+        var sumCarbs = _.reduce(_.map(carbs, function(d) {
+          if (d.normalTime >= start && d.normalTime <= end) {
+            return d.carbInput;
+          }
+          return 0;
+        }), function(total, carbs) {
+          return total + carbs;
+        });
+
+        return {
+          averageDailyCarbs: sumCarbs/((Date.parse(end) - Date.parse(start))/constants.MS_IN_DAY)
```

Comment:
Was there any discussion of how this will be skewed if there's a pump vacation, like the basal:bolus ratio and TDD? We don't show those if there are any days (excepting the most recent) that don't have bolus data (see datamunger lines 109-128).